### ISSUE TITLE
[homematic] Connection timeout issues when upgrading to openhab3 using CCU2

### ISSUE THREAD

**Issue by mstroeve** [openhab/openhab-addons#9855]
I'am using 2.5.X series (currently 2.5.8) flawlessly with homematic since a year. 
Wanted to test the new openhab3 but i'am getting Connection timeout issues on the bridge thing.

I first looked at the binding docs and saw mentions about setting xml-api access right.  
I checked this and it seems right (it also works with the 2.5.X image).
Both settings are stating full access. I also added the Ip address from the openhab host instance just to be sure.

![Schermafbeelding 2021-01-17 om 19 23 21](https://user-images.githubusercontent.com/13969254/104853311-40fb1180-5900-11eb-8d17-efcb2620ede9.png)

Log on the CCU2:
```
Jan 17 19:55:59 de.eq3.cbcs.legacy.bidcos.rpc.internal.InterfaceInitializer INFO  [vert.x-worker-thread-2] Added InterfaceId: HMIP-techniekruimte 
Jan 17 19:56:00 de.eq3.ccu.virtualdevice.service.internal.rega.BackendWorker INFO  [vert.x-worker-thread-11] Execute BackendCommand: de.eq3.ccu.virtualdevice.service.internal.rega.BackendUpdateDevicesCommand 
Jan 17 19:56:05 de.eq3.cbcs.legacy.bidcos.rpc.LegacyServiceHandler INFO  [vert.x-worker-thread-4] (un)registerCallback on LegacyServiceHandler called from url: http://openhab.validation:1000 
Jan 17 19:56:05 de.eq3.ccu.virtualdevice.service.internal.rega.VirtualDeviceHandlerRega INFO  [vert.x-eventloop-thread-4] (un)registerCallback on VirtualDeviceHandlerRega called from url: http://openhab.validation:1000 
Jan 17 19:57:03 de.eq3.cbcs.legacy.bidcos.rpc.internal.InterfaceInitializer ERROR [vert.x-worker-thread-2] IO Exception: Could not add interface: HMIP-techniekruimte 
de.eq3.cbcs.legacy.communication.rpc.RpcIOException: java.net.ConnectException: Connection timed out (Connection timed out)
	at de.eq3.cbcs.legacy.communication.rpc.internal.transport.http.HttpTransport.sendRequest(HttpTransport.java:110)
	at de.eq3.cbcs.legacy.communication.rpc.internal.rpc.RpcClient.sendRequest(RpcClient.java:94)
	at de.eq3.cbcs.legacy.communication.rpc.internal.rpc.RpcClient.invoke(RpcClient.java:82)
	at com.sun.proxy.$Proxy19.listDevices(Unknown Source)
	at de.eq3.cbcs.legacy.bidcos.rpc.internal.LegacyBackendClient.listDevices(LegacyBackendClient.java:139)
	at de.eq3.cbcs.legacy.bidcos.rpc.internal.DeviceUtil.synchronizedBackendDevices(DeviceUtil.java:148)
	at de.eq3.cbcs.legacy.bidcos.rpc.internal.InterfaceInitializer.handle(InterfaceInitializer.java:109)
	at de.eq3.cbcs.legacy.bidcos.rpc.internal.InterfaceInitializer.handle(InterfaceInitializer.java:26)
	at io.vertx.core.eventbus.impl.HandlerRegistration.deliver(HandlerRegistration.java:212)
	at io.vertx.core.eventbus.impl.HandlerRegistration.handle(HandlerRegistration.java:191)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$deliverToHandler$3(EventBusImpl.java:505)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
	at io.vertx.core.impl.TaskQueue.lambda$new$0(TaskQueue.java:60)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection timed out (Connection timed out)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:120)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:179)
	at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:144)
	at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:134)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:612)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:447)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:884)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:55)
	at de.eq3.cbcs.legacy.communication.rpc.internal.transport.http.HttpTransport.sendRequest(HttpTransport.java:106)
	... 15 more


```
When I switched back to the 2.5.8 docker-compose version. The bridge comes up nicely .
Somehow there are differences in the binding version regarding the connection to the bridge?

Environment:
* Docker-compose on Synology.
* CCU with newest 2.55 firmware 
   I also tried the older 2.28 version which was first on the CCU but updating it didn't solve it.

**Comment by MHerbst** [comment]
The problem is that the CCU can not reach the binding. Please check your docker and firewall configuration for the OH3 and make sure that ports 9125 and 9126 are reachable from the CCU. 

This is probably not an issue of the binding (the binding versions 2.5 and 3.0 are identical to a large degree). BTW: Please use the forum for questions like this.

**Comment by mstroeve** [comment]
@MHerbst you are right it might have been better to first try the forum my bad.
The docker config was exactly the same (only different openhab version image). 
Thats why I tried here 😄 first.

It seems the callback ip isn't correctly detected in 3.0 
When I added this in the thing configuration it works like it did in 2.5 without setting this explicitly. 
So there has been some change in 3.0 core or add-on that setting this field is necessary when using docker.
![Schermafbeelding 2021-01-17 om 21 43 51](https://user-images.githubusercontent.com/13969254/104855501-75c19580-590d-11eb-9812-f37be9bf944e.png)
